FrontEnd/myapp/src/buysell.js

```diff
@@ -61,6 +61,10 @@ class Buy extends React.Component {
   async handleSubmit(event) {
     event.preventDefault();
     if (this.state.name != "" && this.state.numberOfShares != null) {
+      if (isNaN(this.state.numberOfShares) || this.state.numberOfShares <= 0) {
+        alert("Please enter a positive number.");
+        return;
+      }
```

Comment:
Maybe we can split it so it gives one message for negative numbers and one for not valid entries? All good either way.